apps/sms/views/inbox/js/inbox.js

```diff
@@ -973,7 +984,21 @@ var InboxView = {
     }
   },
 
-  onDraftSaved: function inbox_onDraftSaved() {
+  onDraftDeleted: function inbox_onDraftDeleted(draft) {
```

Comment:
I see if we just check for thread existence (if there is no thread/draft in Threads then we should not do anything) in "onDraftDeleted" then we won't have any duplicated update/removal - does it work for you? I added 3rd commit with the change, I had to change order in which thread and drafts are removed from respective storages.